oauth/oauth_config_test.go

```diff
@@ -0,0 +1,480 @@
+			TokenNamespace:      secretNamespace,
+			ServiceProviderType: config.ServiceProviderTypeGitHub,
+		}
+
+		oauthCfg, err := ctrl.obtainOauthConfig(ctx, oauthState)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, oauthCfg)
+		assert.Equal(t, oauthCfg.ClientID, "testclientid")
+		assert.Equal(t, oauthCfg.ClientSecret, "testclientsecret")
+		assert.Equal(t, oauthCfg.Endpoint.AuthURL, github.Endpoint.AuthURL)
+		assert.Equal(t, oauthCfg.Endpoint.TokenURL, github.Endpoint.TokenURL)
+		assert.Equal(t, oauthCfg.Endpoint.AuthStyle, github.Endpoint.AuthStyle)
+		assert.Contains(t, oauthCfg.RedirectURL, "baseurl")
+	})
+
+	t.Run("found invalid oauth config secret", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		utilruntime.Must(v1.AddToScheme(scheme))
+		ctx := context.TODO()
+
+		secretNamespace := "test-secretConfigNamespace"
+
+		cl := fake.NewClientBuilder().WithScheme(scheme).WithLists(&v1.SecretList{
+			Items: []v1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "oauth-config-secret",
+						Namespace: secretNamespace,
+						Labels: map[string]string{
+							oauthCfgSecretLabel: string(config.ServiceProviderTypeGitHub),
+						},
+					},
+					Data: map[string][]byte{
+						oauthCfgSecretFieldClientId: []byte("testclientid"),
+					},
+				},
+			},
+		}).Build()
+
+		ctrl := commonController{
+			Config: config.ServiceProviderConfiguration{
+				ClientId:               "eh?",
+				ClientSecret:           "bleh?",
+				ServiceProviderType:    config.ServiceProviderTypeGitHub,
+				ServiceProviderBaseUrl: "http://bleh.eh",
+			},
+			K8sClient: cl,
+			Endpoint:  github.Endpoint,
+			BaseUrl:   "baseurl",
+		}
+
+		oauthState := &oauthstate2.OAuthInfo{
+			TokenNamespace:      secretNamespace,
+			ServiceProviderType: config.ServiceProviderTypeGitHub,
+		}
+
+		oauthCfg, err := ctrl.obtainOauthConfig(ctx, oauthState)
+
+		assert.Error(t, err)
+		assert.Nil(t, oauthCfg)
+	})
+
+	t.Run("error when failed kube request", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		utilruntime.Must(v1.AddToScheme(scheme))
+		ctx := context.TODO()
+
+		secretNamespace := "test-secretConfigNamespace"
+
+		cl := fake.NewClientBuilder().WithScheme(scheme).WithObjectTracker(&mockTracker{
+			listImpl: func(gvr schema.GroupVersionResource, gvk schema.GroupVersionKind, ns string) (runtime.Object, error) {
+				return nil, errors.NewBadRequest("nenenene")
+			}}).Build()
+
+		ctrl := commonController{
+			K8sClient: cl,
+		}
+
+		oauthState := &oauthstate2.OAuthInfo{
+			TokenNamespace:      secretNamespace,
+			ServiceProviderType: config.ServiceProviderTypeGitHub,
+		}
+
+		oauthCfg, err := ctrl.obtainOauthConfig(ctx, oauthState)
+
+		assert.Error(t, err)
+		assert.Nil(t, oauthCfg)
+	})
+}
+
+type mockTracker struct {
+	listImpl func(gvr schema.GroupVersionResource, gvk schema.GroupVersionKind, ns string) (runtime.Object, error)
+}
+
+func (t *mockTracker) Add(obj runtime.Object) error {
+	//TODO implement me
```

Comment:
These are not real TODOs, IMHO... We just don't need these methods yet in any of the tests, right?

Reply:
yes